### ISSUE TITLE
Recursion error in Twarc

### ISSUE THREAD

**Issue by justinlittman** [gwu-libraries/sfm-ui#643]
See http://gwsfm-prod2.wrlc.org/ui/harvests/57674/

The final error is `maximum recursion depth exceeded`.

```
root@gwsfm-prod2:/var/lib/docker/containers/87e9da2ae7cb294b47ada833f1614155917f11a81a76e032977d773406c97a6e# tail -1500 87e9da2ae7cb294b47ada833f1614155917f11a81a76e032977d773406c97a6e-json.log | less

{"log":"    resp = f(*args, **kwargs)\n","stream":"stderr","time":"2017-01-17T13:28:07.811458299Z"}
{"log":"  File \"/usr/local/lib/python2.7/site-packages/twarc.py\", line 312, in new_f\n","stream":"stderr","time":"2017-01-17T13:28:07.811461037Z"}
{"log":"    return f(self, *args, **kwargs)\n","stream":"stderr","time":"2017-01-17T13:28:07.811466158Z"}
{"log":"  File \"/usr/local/lib/python2.7/site-packages/twarc.py\", line 328, in new_f\n","stream":"stderr","time":"2017-01-17T13:28:07.811468832Z"}
{"log":"    return f(self, *args, **kwargs)\n","stream":"stderr","time":"2017-01-17T13:28:07.811471714Z"}
{"log":"  File \"/usr/local/lib/python2.7/site-packages/twarc.py\", line 343, in new_f\n","stream":"stderr","time":"2017-01-17T13:28:07.811474557Z"}
{"log":"    return f(self, *args, **kwargs)\n","stream":"stderr","time":"2017-01-17T13:28:07.811477477Z"}
{"log":"  File \"/usr/local/lib/python2.7/site-packages/twarc.py\", line 643, in get\n","stream":"stderr","time":"2017-01-17T13:28:07.811480253Z"}
{"log":"    r = self.get(*args, **kwargs)\n","stream":"stderr","time":"2017-01-17T13:28:07.811483118Z"}
{"log":"  File \"/usr/local/lib/python2.7/site-packages/twarc.py\", line 271, in new_f\n","stream":"stderr","time":"2017-01-17T13:28:07.811485808Z"}
{"log":"    resp = f(*args, **kwargs)\n","stream":"stderr","time":"2017-01-17T13:28:07.811488653Z"}
{"log":"  File \"/usr/local/lib/python2.7/site-packages/twarc.py\", line 312, in new_f\n","stream":"stderr","time":"2017-01-17T13:28:07.81149135Z"}
{"log":"    return f(self, *args, **kwargs)\n","stream":"stderr","time":"2017-01-17T13:28:07.81149426Z"}
{"log":"  File \"/usr/local/lib/python2.7/site-packages/twarc.py\", line 328, in new_f\n","stream":"stderr","time":"2017-01-17T13:28:07.811496962Z"}
{"log":"    return f(self, *args, **kwargs)\n","stream":"stderr","time":"2017-01-17T13:28:07.811499839Z"}
{"log":"  File \"/usr/local/lib/python2.7/site-packages/twarc.py\", line 343, in new_f\n","stream":"stderr","time":"2017-01-17T13:28:07.81150254Z"}
{"log":"    return f(self, *args, **kwargs)\n","stream":"stderr","time":"2017-01-17T13:28:07.8115054Z"}
{"log":"  File \"/usr/local/lib/python2.7/site-packages/twarc.py\", line 643, in get\n","stream":"stderr","time":"2017-01-17T13:28:07.8115083Z"}
{"log":"    r = self.get(*args, **kwargs)\n","stream":"stderr","time":"2017-01-17T13:28:07.811511306Z"}
{"log":"  File \"/usr/local/lib/python2.7/site-packages/twarc.py\", line 271, in new_f\n","stream":"stderr","time":"2017-01-17T13:28:07.811514276Z"}
{"log":"    resp = f(*args, **kwargs)\n","stream":"stderr","time":"2017-01-17T13:28:07.81151722Z"}
{"log":"  File \"/usr/local/lib/python2.7/site-packages/twarc.py\", line 312, in new_f\n","stream":"stderr","time":"2017-01-17T13:28:07.811520306Z"}
{"log":"    return f(self, *args, **kwargs)\n","stream":"stderr","time":"2017-01-17T13:28:07.811523264Z"}
```

and the tail end:

```
:07.813879718Z"}
{"log":"    unicode(r.url), unicode(r.method), None, r.headers)\n","stream":"stderr","time":"2017-01-17T13:28:07.813882728Z"}
{"log":"  File \"/usr/local/lib/python2.7/site-packages/oauthlib/oauth1/rfc5849/__init__.py\", line 257, in sign\n","stream":"stderr","time":"2017-01-17T13:28:07.81388563Z"}
{"log":"    encoding=self.encoding)\n","stream":"stderr","time":"2017-01-17T13:28:07.81388866Z"}
{"log":"  File \"/usr/local/lib/python2.7/site-packages/oauthlib/common.py\", line 380, in __init__\n","stream":"stderr","time":"2017-01-17T13:28:07.813891594Z"}
{"log":"    self.headers = CaseInsensitiveDict(encode(headers or {}))\n","stream":"stderr","time":"2017-01-17T13:28:07.813894676Z"}
{"log":"  File \"/usr/local/lib/python2.7/site-packages/oauthlib/common.py\", line 376, in \u003clambda\u003e\n","stream":"stderr","time":"2017-01-17T13:28:07.813900101Z"}
{"log":"    encode = lambda x: to_unicode(x, encoding) if encoding else x\n","stream":"stderr","time":"2017-01-17T13:28:07.81390348Z"}
{"log":"  File \"/usr/local/lib/python2.7/site-packages/oauthlib/common.py\", line 312, in to_unicode\n","stream":"stderr","time":"2017-01-17T13:28:07.813906426Z"}
{"log":"    dict(data)\n","stream":"stderr","time":"2017-01-17T13:28:07.813909705Z"}
{"log":"  File \"/usr/local/lib/python2.7/_abcoll.py\", line 410, in keys\n","stream":"stderr","time":"2017-01-17T13:28:07.813912634Z"}
{"log":"    return list(self)\n","stream":"stderr","time":"2017-01-17T13:28:07.813915552Z"}
{"log":"RuntimeError: maximum recursion depth exceeded\n","stream":"stderr","time":"2017-01-17T13:28:07.813918316Z"}
```

**Comment by justinlittman** [comment]
What I think is happening is that a connection error is occurring.  This causes get() to retry.  But when get() retries, it doesn't pass the allow_404 and connection_error_count kwargs.  When the get is for a missing user lookup (which should trigger an allowed 404), this causes a recursion.

**Comment by justinlittman** [comment]
This is fixed by Twarc 1.0.4.  Upgrading is included in https://github.com/gwu-libraries/sfm-twitter-harvester/tree/t579-stop_twitter, which is part of #579.
